File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.kt
class Solution {
    fun zeroFilledSubarray(nums: IntArray): Long {
        var cnt:Long = 0
        var ans:Long = 0
        
        nums.forEach{
            n ->
            if(n == 0){
                cnt++
            }
            else{
                ans += (cnt * (cnt+1) )/2 
                cnt = 0
            }
        }
        
        if (cnt > 0) {
        ans += (cnt * (cnt+1) )/2 }
        
        return ans
    }
}

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        cnt = 0
        ans = 0
        for n in nums:
            if n:
                ans += (cnt * (cnt+1))//2
                cnt = 0
            else:
                cnt += 1
        if cnt:
            ans += (cnt * (cnt+1))//2
        return ans
            
        